test/e2e/controller/missing_test.go

```diff
@@ -75,8 +75,67 @@ func deleteAgonesControllerPods() error {
 	return nil
 }
 
+func waitForAgonesControllerRunning() error {
+	return wait.PollImmediate(time.Second, 5*time.Minute, func() (bool, error) {
+		list, err := getAgonesControllerPods()
+		if err != nil {
+			return true, err
+		}
+
+		for i := range list.Items {
+			for _, c := range list.Items[i].Status.ContainerStatuses {
+				if c.State.Running == nil {
+					return false, nil
+				}
+			}
+		}
+
+		return true, nil
+	})
+}
+
 // getAgonesControllerPods returns all the Agones controller pods
 func getAgonesControllerPods() (*corev1.PodList, error) {
 	opts := metav1.ListOptions{LabelSelector: labels.Set{"agones.dev/role": "controller"}.String()}
 	return framework.KubeClient.CoreV1().Pods("agones-system").List(opts)
 }
+
+func defaultGameServer(namespace string) *agonesv1.GameServer {
```

Comment:
Can we refactor [this](https://github.com/googleforgames/agones/blob/master/test/e2e/gameserver_test.go#L577) instead?

Reply:
Yeah I thought about centralising that too - moving it into framework maybe. Wasn't sure if 2 duplication was enough to warrant it. If you think so, I'm happy to do it.

I assume that is what you mean here?

Reply:
Correct, thanks for the clarification. framework.go should be a good place or you can add a new file such as testhelper and centralize the common helper methods there?

Reply:
Sounds good - I like the idea of testhelper.go or similar (I think we have a previous pattern for this somewhere, maybe, I can copy).